src/views/Applications/ApplicationForm.vue

```diff
@@ -287,7 +312,10 @@ export default defineComponent({
         !currentState.value.matches('pending') &&
         formData.value.name.length &&
         (appRegV2Enabled && formMode.value !== 'edit' ? hasAppAuthStrategies.value : true) &&
-        (isDcr.value ? true : formData.value.reference_id?.length)
+        (appRegV2Enabled
+          ? (appIsDcr.value ? true : formData.value.reference_id?.length)
+          : (isDcr.value ? true : formData.value.reference_id?.length)
```

Comment:
[nit] Since `appIsDcr` is a boolean ref, you could do 

```suggestion
          ? (appIsDcr.value || formData.value.reference_id?.length)
          : (isDcr.value || formData.value.reference_id?.length)
```